tweets/views.py

```diff
@@ -1,9 +1,55 @@
-from django.contrib.auth.mixins import LoginRequiredMixin
-from django.views.generic import TemplateView
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.urls import reverse_lazy
+from django.views.generic import CreateView, DeleteView, DetailView, ListView
 
+from .forms import TweetForm
+from .models import Tweet
 
-class HomeView(LoginRequiredMixin, TemplateView):
+# , TemplateView
+
+
+# class HomeView(LoginRequiredMixin, TemplateView):
+# template_name = "tweets/home.html"
+
+
+class HomeView(LoginRequiredMixin, ListView):
+    model = Tweet
     template_name = "tweets/home.html"
+    ordering = "-created_at"
+    # model = Tweet
+    # template_name = "tweets/home.html"
+    # queryset = model.objects.select_related("user")

```

Comment:
```suggestion
    queryset = model.objects.select_related("user")
```
のようにコメント化を解除することで最適化できると思います。